src/backend/postmaster/pgstat.c

```diff
@@ -2913,6 +2915,10 @@ gpstat_report_waiting(char reason)
 	 * protocol.  The update must appear atomic in any case.
 	 */
 	beentry->st_waiting = reason;
+	if (reason == PGBE_WAITING_NONE)
+		beentry->st_waiting_start_timestamp = 0;
+	else
+		beentry->st_waiting_start_timestamp = GetCurrentTimestamp();
```

Comment:
Does set up timestamp just regard to waiting on locks? Here we track the starting timestamp for all wait events. So commit message and PR description have to be revised.

Reply:
Fixed. Thanks.